.github/workflows/destroy-pr-infra.yml

```diff
@@ -0,0 +1,56 @@
+name: Pull request
+on:
+  pull_request:
+    types: [closed]
```

Comment:
I think the good practice is to add `workflow_dispatch` trigger to such workflows. You can add `input` with PR number so then you can manually run destroy of PR infra for selected PR.